### PR TITLE
Feature/apiIntegration

### PR DESCRIPTION
**Pull request by D-ote**
setup request processor
setup dispatch in loginAction
setup useSelector in loginPage
setup notification manager

### CODE REVIEW

File: src/components/loginPage.jsx [hunk 1]
```diff
@@ -54,12 +45,12 @@ const LoginPage = () => {
           <input
             type="email"
             name="email"
-            value={email}
+            value={loginDetails?.email}
```
Comment (Dverybest) [inline src/components/loginPage.jsx]: Optional chaining is not necessary here... remove '?'
```suggestion
            value={loginDetails.email}
```

File: src/components/loginPage.jsx [hunk 2]
```diff
@@ -69,12 +60,12 @@ const LoginPage = () => {
           <input
             type="password"
             name="password"
-            value={password}
+            value={loginDetails?.password}
```
Comment (Dverybest) [inline src/components/loginPage.jsx]: remove optional chaining... refactor to
```suggestion
            value={loginDetails.password}
```

File: src/components/loginPage.jsx [hunk 3]
```diff
@@ -1,40 +1,31 @@
...
-        } else {
+  const handleChange=(e)=>{
+    setLoginDetails({...loginDetails, [e.target.id]: e.target.value})
+  }
+  
+  useEffect(() => {
+    if(auth?.profile?.authToken){
+      history.replace("/dashboard")
+    }
+  }, [auth, history])
```
Comment (Dverybest) [inline src/components/loginPage.jsx:24]: Why are we listening for change in history?

**Review by Dverybest — CHANGES_REQUESTED** [review]

File: src/components/loginPage.jsx (hunk 3 above)
Comment (D-ote) [reply src/components/loginPage.jsx:24]: I don't know. React requested it.

**Review by Dverybest — APPROVED** [review]

**Pull request merged**